### ISSUE TITLE
"instrument access failed" with Spyder X pro

### ISSUE THREAD

**Issue by Mister-Teatime** [eoyilmaz/displaycal-py3#249]
I just bought a Spyder X pro, and it is recognized on my machine (yay), black point calibration seems to work, but directly after that, I'm getting a message window with the error above.

I saw issue  #143, and it looks pretty much like that (did not test that script, though).

What I did test: I tried to run a calibration on my laptop, and it worked -- the first time around. I didn't like the result, so I chose another white point, and when the second run was about to start, just after blackpoint calibration, the same issue started happening. From then on, anything I did cause the same error message. Unplugging the sensor and plugging it back in, even into another USB port, does not help.

Back on my PC, even after rebooting and trying a bunch of different USB ports, the issue seems to be persistent: Whatever the first operation is works, and then it fails. I also remembered the mainboard manual for my PC mentioning that some USB ports work differently than others, so I tried one of each "type", including the USB 2.0 ports usually used for mouse and keyboard, which did not seem to make any difference.

**Technical details:**
PC:
OS: Manjaro, Linux kernel 5.15.108, Plasma 5.27.4
Display: Tried both Wayland and X11 -- found no difference
Graphics: AMD Radeon RX 5500 XT

Laptop: Dell Inspiron 15 7000 2-in-1
OS Manjaro, Linux Kernel 6.1.25, Plasma 5.27.4
Display: X11
Graphics Intel UHD

The laptop can also dual-boot Windows, but I have not had time yet to set DisplayCal up there. Be happy to do that if it's helpful in figuring out what's going on.

I'm not sufficiently versed to know what kind of other information may be relevant and useful in this matter, so please let me know. Manjaro lets me switch LTS-kernels from 4.19 to 6.1, so I could also test those if you think that could help.

**Comment by eoyilmaz** [comment]
It seems that there is an update from ArgyllCMS 2.2.1 to 2.3.0 related to that, can you check if you are using ArgyllCMS 2.3.0.

And here is the page I found that info: https://hub.displaycal.net/forums/topic/error-instrument-access-failed-for-lcd-white-led-mode/

**Comment by eoyilmaz** [comment]
Another info from that same page, it seems Argyll 2.3.1 has the same failure while 2.3.0 is working fine for people.

**Comment by Mister-Teatime** [comment]
Thanks for the hint!
I indeed have ArgyllCMS 2.3.1-3 installed, and when I switched to RGB-LED, I was able to measure my monitor.

However, when I then activated the ArgyllCMS debugging output (no idea if it would have told me anything useful...) and re-ran without any changes, I got the "access failed" problem again. Switched through all backlight modes, and turned the debugging output off again, but now it's stuck again.
Restarting DisplayCal does not restore it, either.

I guess I'd have to downgrade to 2.3.0, then? Just tried doing that via Manjaro's ``downgrade`` command, and it turns out there are only different revisions of 2.3.1  in the repo, no 2.3.0 anymore ... will try a userspace installation, but unlikely I can find the time before the weekend.

**Comment by eoyilmaz** [comment]
you don't need to install it actually, I think you can just download and extract the tarball here: https://www.argyllcms.com/Argyll_V2.3.0_linux_x86_64_bin.tgz

and show the bin folder from where you extracted the tarball to DisplayCAL.

**Comment by Mister-Teatime** [comment]
The solution turned out to be easier than I thought:

This post on the ArgyllCMS mailing list describes the problem:
[https://www.freelists.org/post/argyllcms/SpyderX-on-231-Linux](https://www.freelists.org/post/argyllcms/SpyderX-on-231-Linux)
(TLDR: The author made a mistake, and the 2.3.1 update reverted some bugfixes by accident)

A bit later in the thread, they suggest to replace some of the files directly with versions from the upcoming update, and provide a download link to the files in question:
[https://www.freelists.org/post/argyllcms/SpyderX-on-231-Linux,2](https://www.freelists.org/post/argyllcms/SpyderX-on-231-Linux,2)

I downloaded the archive, extracted the contents into `/usr/bin`, and one reboot later the error is gone.
I'm still seeing other errors in some circumstances, but I was able to calibrate and profile my display, multiple times, with different settings, and have not seen the "access failed" message again.

(Also, I am happy to report that the Spyder X is in fact *way* faster than my very old Spyder1. Calibration used to take over 2 hours, now it's done in less than 15 minutes.)
